src/core/dynamic_batch_scheduler.cc

```diff
@@ -198,60 +155,44 @@ DynamicBatchScheduler::Enqueue(std::unique_ptr<InferenceRequest>& request)
     // Assuming no error is returned, this call takes ownership of
     // 'request' and so we can't use it after this point.
     RETURN_IF_ERROR(queue_.Enqueue(request->Priority(), request));
-
-    // If there are any idle runners and the queued batch size is greater or
-    // equal to next preferred batch size, then wake one up to service this
-    // request. We do the actual wake outside of the lock to avoid having the
-    // woken thread immediately block on the lock
-    wake_runner = (idle_scheduler_thread_cnt_ > 0);
```

Comment:
I think you want to keep this logic. We don't want to wakeup the scheduler thread unless these conditions are true.

Reply:
We will not hit this logic as there is only a single scheduler thread in the system. 
For multiple instances, if a request is being allocated to the last available BackendThread, then I will block the scheduler thread just after the hand off till a BackendThread returns from execution.

Reply:
I think it may be better to not block the scheduler thread. There is work for the scheduler thread to do as requests come in and it would be good to allow it do continue doing that work even if no instances are available. The rate limiter will make sure that the scheduler thread does not call back into the schedule/execute function until an instance is actually available.